SU2_CFD/src/drivers/CDriver.cpp

```diff
@@ -64,310 +62,98 @@ CDriver::CDriver(char* confFile,
-                           config_container[iZone], iInst);
-
-    } // End of loop over iInst
-
-    if (rank == MASTER_NODE)
-      cout << endl <<"----------------- Integration and Numerics Preprocessing ----------------" << endl;
-
-    /*--- Definition of the integration class: integration_container[#ZONES][#INSTANCES][#EQ_SYSTEMS].
-     The integration class orchestrates the execution of the spatial integration
-     subroutines contained in the solver class (including multigrid) for computing
-     the residual at each node, R(U) and then integrates the equations to a
-     steady state or time-accurately. ---*/
-
+    /*--- Read the number of instances for each zone ---*/
+    
+    nInst[iZone] = config_container[iZone]->GetnTimeInstances();
+    
+    geometry_container[iZone]    = new CGeometry**    [nInst[iZone]];
+    iteration_container[iZone]   = new CIteration*    [nInst[iZone]];
+    solver_container[iZone]      = new CSolver***     [nInst[iZone]];
     integration_container[iZone] = new CIntegration** [nInst[iZone]];
-    for (iInst = 0; iInst < nInst[iZone]; iInst++){
-      integration_container[iZone][iInst] = NULL;
-
-      integration_container[iZone][iInst] = new CIntegration*[MAX_SOLS];
-      Integration_Preprocessing(integration_container[iZone], geometry_container[iZone],
-          config_container[iZone], iInst);
+    numerics_container[iZone]    = new CNumerics****  [nInst[iZone]]; 
+    grid_movement[iZone]         = new CVolumetricMovement* [nInst[iZone]];
+    
+    /*--- Allocate transfer and interpolation container --- */
+    
+    transfer_container[iZone]     = new CTransfer*[nZone];
+    interpolator_container[iZone] = new CInterpolator*[nZone];
+    
+    for (jZone = 0; jZone < nZone; jZone++){
+      transfer_container[iZone][jZone]             = NULL;
+      interpolator_container[iZone][jZone]         = NULL;
     }
     
-    if (rank == MASTER_NODE) cout << "Integration Preprocessing." << endl;
-
-    /*--- Definition of the numerical method class:
-     numerics_container[#ZONES][#INSTANCES][#MG_GRIDS][#EQ_SYSTEMS][#EQ_TERMS].
-     The numerics class contains the implementation of the numerical methods for
-     evaluating convective or viscous fluxes between any two nodes in the edge-based
-     data structure (centered, upwind, galerkin), as well as any source terms
-     (piecewise constant reconstruction) evaluated in each dual mesh volume. ---*/
-
-    numerics_container[iZone] = new CNumerics****[nInst[iZone]];
     for (iInst = 0; iInst < nInst[iZone]; iInst++){
-      numerics_container[iZone][iInst] = NULL;
-
-      numerics_container[iZone][iInst] = new CNumerics***[config_container[iZone]->GetnMGLevels()+1];
-
-      Numerics_Preprocessing(numerics_container[iZone], solver_container[iZone],
-          geometry_container[iZone], config_container[iZone], iInst);
+      
+      config_container[iZone]->SetiInst(iInst);
```

Comment:
Is it possible to avoid this way of passing information via weird side effects?

Reply:
I'll have to check, probably it can be avoided.

Reply:
Unfortunately it cannot be avoided for now. Otherwise we have to change the whole Harmonic Balance structure and I don't want to do that at the moment.

Reply:
Fair enough but this stuff should be cleaned up before it becomes standard practice to use the config as a messenger.